source/common/ssl/context_config_impl.cc

```diff
@@ -41,18 +65,32 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::auth::CommonTlsContex
           Config::DataSource::read(config.validation_context().crl(), true)),
       certificate_revocation_list_path_(
           Config::DataSource::getPath(config.validation_context().crl())),
-      cert_chain_(
-          config.tls_certificates().empty()
-              ? ""
-              : Config::DataSource::read(config.tls_certificates()[0].certificate_chain(), true)),
+      cert_chain_([&config, &secret_manager] {
+        if (!config.tls_certificates().empty()) {
+          return Config::DataSource::read(config.tls_certificates()[0].certificate_chain(), true);
+        } else if (!config.tls_certificate_sds_secret_configs().empty()) {
+          auto secret = findSecretAndCheckType<Ssl::TlsCertificateConfigImpl>(
+              secret_manager, config.tls_certificate_sds_secret_configs()[0].name());
+          return (secret) ? secret->certificateChain() : std::string("");
+        } else {
+          return std::string("");
+        }
+      }()),
```

Comment:
IMHO,  it is easier to read if making it as a separate function.

Reply:
+1, please figure out how to break up / simplify the logic in this function.

Reply:
Created separated functions